services/cognito/spec/requests/users_spec.rb

```diff
@@ -31,6 +31,38 @@
         expect_json_sizes('data', models_count)
       end
     end
+
+    context 'perform query' do
+      include_context 'authorized user'
+
+      let(:models_count) { rand(1..5) }
+      let!(:models) { create_list :user, models_count }
+
+      context 'based on ID' do
+        before do
+          get url + "?filter[query]=#{models.last.id}", headers: request_headers
+        end
+
+        it 'returns and ok response with the user ID queried' do
+          expect(response).to have_http_status(:ok)
+          expect_json_sizes('data', 1)
+        end
+      end
+
+      context 'based on primary_identifier' do
+        let(:model_primary_identifier) { models.last.primary_identifier }
+        let(:response_count) { models.select { |model| model.primary_identifier == model_primary_identifier }.length }
```

Comment:
I think it would be interesting that this test would cover the different scenarios. E.g
create a few models where each one matches one of the attributes we are using for filtering, and one where none of the attributes used for the filter will match. Ensure that all of the ones that match are returned properly, while th one that does not match does not return

Reply:
E.g. We are matching against: `primary_identifier` `first_name` `last_name` `email_address`
so, I'd create 5 models:
1. create :user, primary_identifier: 'bananas_123'
2. create :user, first_name: 'Bananas' 
3. create :user, last_name: 'Bananas' 
4. create :user, email: 'amazing_bananas_123@gmail.com' 
5. create :user

and then filter for `ananas` and expect 1 to 4 to be returned